console/console-init/ui/src/modules/address-space/components/EndpointConfiguraion/EndpointConfiguraion.tsx

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import {
+  Grid,
+  GridItem,
+  Form,
+  Checkbox,
+  FormGroup,
+  Radio,
+  Switch
+} from "@patternfly/react-core";
+import { IMessagingProject } from "modules/address-space/dialogs";
+import {
+  tlsCertificateOptions,
+  endpointProtocolOptions,
+  TlsCertificateType
+} from "modules/address-space/utils";
+import { StyleSheet, css } from "@patternfly/react-styles";
+
+const style = StyleSheet.create({
+  margin_left: {
+    marginLeft: 20
+  }
+});
+interface IEndpointConfigurationProps {
+  projectDetail: IMessagingProject;
+  setProjectDetail: (project: IMessagingProject) => void;
+}
+const EndpointConfiguration: React.FunctionComponent<IEndpointConfigurationProps> = ({
+  projectDetail,
+  setProjectDetail
+}) => {
+  const { tlsCertificate, protocols } = projectDetail;
+  const onRouteChange = (value: boolean) => {
+    if (!value) {
+      setProjectDetail({
+        ...projectDetail,
+        tlsTermination: undefined,
+        hostname: undefined,
+        addRoutes: value
+      });
+    } else {
+      setProjectDetail({ ...projectDetail, addRoutes: value });
+    }
+  };
+  const onCertificateChange = (_: boolean, event: any) => {
+    const value = event.target.value;
+    if (value) {
+      if (value === TlsCertificateType.UPLOAD_CERT) {
+        setProjectDetail({
+          ...projectDetail,
+          tlsCertificate: value,
+          addCertificate: true
+        });
+      } else {
+        setProjectDetail({
+          ...projectDetail,
+          tlsCertificate: value,
+          addCertificate: false
+        });
+      }
+    }
+  };
+  const findPrototcol = (protocol: string) => {
+    if (protocols) {
+      const filteredProtocols = protocols.filter(pr => pr === protocol);
+      return filteredProtocols[0];
+    }
+  };
+  const isProtocolPresent = (protocol: string) => {
+    if (protocols) {
```

Comment:
can we try single if condition, we dont need second one.
if(protocols && findPrototcol(protocol)){
  return true;
}
return false;